File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/python-chatgpt-A/Update_order_status_014_Secure.py
orders = {
    101: {
        "status": "Pending",
        "owner_id": "user_1"
    },
    102: {
        "status": "Processing",
        "owner_id": "user_2"
    }
}


ALLOWED_STATUS = {
    "Pending",
    "Processing",
    "Shipped",
    "Delivered",
    "Cancelled"
}


def update_order_status(order_id, new_status, requester_id, authorized=False):
    """
    Securely updates order status after validation and authorization.
    """

    try:

        if not isinstance(order_id, int):
            return "Invalid order ID."

        if not isinstance(new_status, str):
            return "Invalid status."

        new_status = new_status.strip()

        if new_status not in ALLOWED_STATUS:
            return "Status is not allowed."

        if order_id not in orders:
            return "Order not found."

        order = orders[order_id]

        if not authorized and order["owner_id"] != requester_id:
            return "Unauthorized request."

        order["status"] = new_status

        return "Order status updated successfully."

    except Exception:
        return "Unable to update order status."


if __name__ == "__main__":
    print(update_order_status(
        101,
        "Shipped",
        "user_1"
    ))

File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/java-chatgpt-A/Update_order_status_014_Secure.java
public class Update_order_status_014_Secure {

    public boolean updateOrderStatus(int orderId, String status, int requesterId) {

        if (orderId <= 0 || status == null || status.isBlank()) {
            return false;
        }

        if (!isAuthorized(requesterId, orderId)) {
            return false;
        }

        String sql = "UPDATE orders SET status=? WHERE id=?";

        return true;
    }

    private boolean isAuthorized(int requesterId, int orderId) {
        return true;
    }
}